<launch>

    <param name="use_sim_time" value="true"/>
    <arg name="need_map" default="true"/> 
    <arg name="rviz" default="true" /> 
    <arg name="bag_file_tongo" default="$(env HOME)/Documents/Bags/teatro/last_day/casi_2023-03-06-10-21-26_edited.bag" />


    <arg name="bag_file" default="$(env HOME)/Documents/Bags/teatro/last_day/refinitiva_2023-03-06-12-19-25.bag" />
    <arg name="mission_file" default="teatro_wall_opt" /> 
    <arg name="mission_path" default="$(find marsupial_mission_interface)/cfg/optimized_path/optimized_path_$(arg mission_file).yaml"/>
    <arg name="statistical_results_path" default="$(env HOME)/statistical_results_from_validation_experiments_marsupial/"/>
    <arg name="map_name" default="teatro_wall"/> 
    <arg name="ugv_base_frame" default="arco/base_link" /> 
    <arg name="uav_base_frame" default="base_link" /> 
    <arg name="world_frame"     default="world" /> 
    <arg name="interpolation_distance" default="0.5"/>

    <node name="get_data_error_mission_node" pkg="marsupial_mission_interface" type="get_data_error_mission_node" output="screen"># launch-prefix="gdb -ex run --args">
        <param name="mission_path" value="$(arg mission_path)"/> 
        <param name="statistical_results_path" value="$(arg statistical_results_path)"/> 
        <param name="map_name" value="$(arg map_name)"/> 
        <param name="ugv_base_frame" value="$(arg ugv_base_frame)"/> 
        <param name="uav_base_frame" value="$(arg uav_base_frame)"/> 
        <param name="world_frame" value="$(arg world_frame)"/> 
        # Here parameter to load mission
        <rosparam command="load" file="$(env HOME)/marsupial_ws/src/marsupial_mission_interface/cfg/offset_maps_dll/offset_maps_dll_$(arg map_name).yaml"/>
        <param name="interpolation_distance" value="$(arg interpolation_distance)"/> 
        <param name="vel_ugv" value="0.3"/> 
        <param name="vel_uav" value="0.2"/>
    </node>

    # For Bag: refinitiva_2023-03-06-12-19-25.bag from 83 [s]

    #To show MAP from DLL and not affect TFs
    <include file="$(find marsupial_launchers)/arco_launch/dll_arco.launch" if="$(arg need_map)">
        <arg name="map_path" value="$(env HOME)/marsupial_ws/src/marsupial_optimizer/maps/$(arg map_name).bt"/>    
        <arg name="odom_frame_id" value="arco/odom_tongo"/>
        <arg name="base_frame_id" value="arco/base_link_tongo"/>
    </include>
    <node name="bag_player_tongo" pkg="rosbag" type="play" args="$(arg bag_file_tongo) --clock" if="$(arg need_map)">
        <remap from="/tf" to="/tf_tongo"/>
        <remap from="/tf_static" to="/tf_static_tongo"/>
        <remap from="/arco_os1_cloud_node/points_non_dense" to="/arco_os1_cloud_node/points_non_dense_tongo"/>
    </node>
    <node pkg="rviz" name="rviz" type="rviz" args="-d $(find marsupial_mission_interface)/viz/check_mission.rviz" if="$(arg rviz)"/>

</launch>